### ISSUE TITLE
type object "SomeBravadoResourceType" has no attribute 'marshal' ( and 'unmarshal')

### ISSUE THREAD

**Issue by adamhadani** [Yelp/bravado-core#160]
Started seeing these errors today after installing the just release 4.7.0 bravado-core. Downgrading to previous version (4.6.1) makes these go away so seems like some kind of regression.. Here's an example snippet:

```python
...
chatroom_event.marshal(),
  File "/Users/adamhadani/.virtualenvs/venv/lib/python3.6/site-packages/bravado_core/model.py", line 176, in __getattr__
    .format(type(self).__name__, attr_name)
AttributeError: type object 'SomeBravadoResourceType' has no attribute 'marshal'
```

Also, seeing similar error when trying to use `unmarshal`:

```python
...
.unmarshal({
AttributeError: type object 'SomeBravadoResourceType' has no attribute 'unmarshal'
```

**Comment by adamhadani** [comment]
FYI @sjaensch

**Comment by macisamuele** [comment]
@adamhadani you're right ... version 4.7.0 changes the way to marshal and unmarshal models.
Instead of using ``marshal`` and ``unmarshal`` class methods are replaced by ``_marshal`` and ``_unmarshal`` ones. This was injected by #159.

Since actually this is a back incompatible change we should change the package version and make evident in the changelog how to migrate from ``v4.6.1`` to ``v5.0.0`` (or any version code that will be attributed to the new version)

Thanks a lot for highlighting this ... we'll try to fix this regression soon!

**Comment by adamhadani** [comment]
ping @macisamuele I have a PR  on the way for this which re-introduces original function names with a deprecation warning, will ping u there when up, thanks for confirmation!

**Comment by sjaensch** [comment]
Fixed by #161.
